pkg/catalogv2/content/content.go

```diff
@@ -241,7 +244,23 @@ func (c *Manager) Chart(namespace, name, chartName, version string, skipFilter b
 		return nil, err
 	}
 
-	return helmhttp.Chart(secret, repo.status.URL, repo.spec.CABundle, repo.spec.InsecureSkipTLSverify, repo.spec.DisableSameOriginCheck, chart)
+	// Check if chart is nil and has atleast one URL
+	if chart == nil {
+		return nil, errors.New("chart is nil")
+	}
+	if len(chart.URLs) <= 0 {
+		return nil, errors.New("chart has no urls specified")
+	}
+
+	switch {
+	// For OCI based helm repositories, there is no index.yaml.
+	// We generate index.yaml for it. While generating the index.yaml
+	// we only set index 0 of chart.URLs.
+	case registry.IsOCI(chart.URLs[0]):
+		return oci.Chart(secret, chart, chart.URLs[0], *repo.spec)
+	default:
+		return helmhttp.Chart(secret, repo.status.URL, repo.spec.CABundle, repo.spec.InsecureSkipTLSverify, repo.spec.DisableSameOriginCheck, chart)
```

Comment:
```suggestion
		return helmhttp.Chart(secret, repo.status.URL, repo.spec.CABundle, repo.spec.InsecureSkipTLSverify, repo.spec.InsecureDisableSameOriginCheck, chart)
```

Reply:
Will change it in another PR since it is a field introduced before.